### PR TITLE
Let language decide test file paths instead of hardcoding that

### LINKED ISSUE #58: Don't hardcode test file path
For LLM models we currently hardcode the test file path. This does not scale.

- [x] let language decide what the test file path is

### PR DESCRIPTION
**Pull request by bauersimon**
Closes #58

### CODE REVIEW

File: language/golang/golang.go [hunk 1]
```diff
@@ -59,6 +59,13 @@ func (l *Language) Files(log *log.Logger, repositoryPath string) (filePaths []st
 	return filePaths, nil
 }
 
+// TestFilePath returns the file path of a test file given the corresponding file path of the test's source file.
+// The file paths may be relative.
+func (l *Language) TestFilePath(filePath string) (testFilePath string) {
```
Comment (zimmski) [inline language/golang/golang.go:64]: Please add the repository path as well. The file path of Java projects is dependent on the projects path.

File: language/golang/golang.go [hunk 2]
```diff
@@ -59,6 +59,13 @@ func (l *Language) Files(log *log.Logger, repositoryPath string) (filePaths []st
 	return filePaths, nil
 }
 
+// TestFilePath returns the file path of a test file given the corresponding file path of the test's source file.
```
Comment (zimmski) [inline language/golang/golang.go:62]: Sounds complicated, why not?

-> `TestFilePath returns the test file path of a implementation file path.`

File: language/golang/golang.go (hunk 1 above)
Comment (bauersimon) [reply language/golang/golang.go:64]: Mhm or we change the docstring that the filepath always needs to be relative to the project root?

Comment (bauersimon) [reply language/golang/golang.go:64]: but just adding the project root is probably fine as well

Comment (zimmski) [reply language/golang/golang.go]: All our parameters are always relative to their context. So would not mention that in the doc

**Review by zimmski — APPROVED** [review]

**Pull request merged**